doc/Internals.md

```diff
@@ -460,5 +460,7 @@ So if you need to trigger events in theia:
  - Register events with the emitter.event function
  - Fire events with emitter.fire(event)
 
+# Preferences
 
+You can check out [this page](Preferences.md) to learn more about how preferences work within Theia.
 
```

Comment:
@simark Yes one file per subject or theme sounds good.

For the extensions doc vs internals however, it seems to me that pretty much everything in Theia is relevant for an extension developer since it's all extensions internally too. So I don't think we need the distinction much... or at least not for now.

So I agree with @epatpol